lib/Myriad/Bootstrap.pm

```diff
@@ -128,35 +175,49 @@ sub boot {
         }
     }
 
-    # Establish comms channel for child process
-    socketpair my $child_pipe, my $parent_pipe, $constant{AF_UNIX}, $constant{SOCK_STREAM}, $constant{PF_UNSPEC}
-        or die $!;
+    my ($inotify_parent_pipe, $inotify_child_pipe) = open_pipe();
 
-    { # Unbuffered writes
-        my $old = select($child_pipe);
-        $| = 1; select($parent_pipe);
-        $| = 1; select($old);
+    {
+        if (my $pid = fork // die "fork! $!") {
+            push @children_pids, $pid;
+        } else {
+            local $SIG{HUP}= sub {
+                say "$pid - inotify process termintated";
+                exit 0;
+            };
+
+            require Module::Load;
+            Module::Load::load('Linux::Inotify2');
+
+            my $watcher = Linux::Inotify2->new();
+            $watcher->blocking(0);
+            while (1) {
+                check_messages_in_pipe($inotify_parent_pipe, sub {
+                    my $module_path = shift;
+                    say "$$ - Going to watch $module_path for changes";
+                    $watcher->watch($module_path, $constant{IN_MODIFY}, sub {
+                        my $e = shift;
+                        print $inotify_parent_pipe "change\r\n";
+                        # The new child might have different paths
+                        $e->w->cancel;
+                    });
+                });
+                $watcher->poll;
+            }
+            exit 0;
+        }
     }
 
+    my ($parent_pipe, $child_pipe) = open_pipe();
```

Comment:
You open this socketpair here once. But you fork several kids to reuse it. I take it there is only one kid at a time. However, what happens if the parent writes something and the kid dies before reading it? The next one then finds something in the pipe although it was meant for the predecessor. 

Maybe it's better to recreate a fresh pair for each kid.